app/src/main/java/fr/free/nrw/commons/upload/structure/depicts/DepictModel.java

```diff
@@ -0,0 +1,208 @@
+import io.reactivex.Observable;
+import io.reactivex.disposables.CompositeDisposable;
+import timber.log.Timber;
+
+public class DepictModel {
+    private static final int SEARCH_DEPICTS_LIMIT = 25;
+    private final DepictDao depictDao;
+    private final DepictsInterface depictsInterface;
+    private final JsonKvStore directKvStore;
+    @Inject
+    GpsDepictsModel gpsDepictsModel;
+    @Inject
+    CategoryClient categoryClient;
+
+    private List<DepictedItem> selectedDepictedItems;
+    private HashMap<String, ArrayList<String>> depictsCache;
+    private CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    @Inject
+    public DepictModel(DepictDao depictDao, @Named("default_preferences") JsonKvStore directKvStore, DepictsInterface depictsInterface) {
+        this.depictDao = depictDao;
+        this.directKvStore = directKvStore;
+        this.depictsInterface = depictsInterface;
+        this.depictsCache = new HashMap<>();
+        this.selectedDepictedItems = new ArrayList<>();
+    }
+
+    public Comparator<DepictedItem> sortBySimilarity(final String filter) {
+        Comparator<String> stringSimilarityComparator = StringSortingUtils.sortBySimilarity(filter);
+        return (firstItem, secondItem) -> stringSimilarityComparator
+                .compare(firstItem.getDepictsLabel(), secondItem.getDescription());
+    }
+
+    public boolean containsYear(String item) {
+        //Check for current and previous year to exclude these categories from removal
+        Calendar now = Calendar.getInstance();
+        int year = now.get(Calendar.YEAR);
+        String yearInString = String.valueOf(year);
+
+        int prevYear = year - 1;
+        String prevYearInString = String.valueOf(prevYear);
+        Timber.d("Previous year: %s", prevYearInString);
+
+        //Check if item contains a 4-digit word anywhere within the string (.* is wildcard)
+        //And that item does not equal the current year or previous year
+        //And if it is an irrelevant category such as Media_needing_categories_as_of_16_June_2017(Issue #750)
+        //Check if the year in the form of XX(X)0s is relevant, i.e. in the 2000s or 2010s as stated in Issue #1029
+        return ((item.matches(".*(19|20)\\d{2}.*") && !item.contains(yearInString) && !item.contains(prevYearInString))
+                || item.matches("(.*)needing(.*)") || item.matches("(.*)taken on(.*)")
+                || (item.matches(".*0s.*") && !item.matches(".*(200|201)0s.*")));
+    }
+
+    public void cacheAll(HashMap<String, ArrayList<String>> categories) {
+        depictsCache.putAll(categories);
+    }
+
+    public HashMap<String, ArrayList<String>> getDepictsCache() {
+        return depictsCache;
+    }
+
+    boolean cacheContainsKey(String term) {
+        return depictsCache.containsKey(term);
+    }
+
+    /*public Observable<DepictedItem> searchAll(String term, List<UploadMediaDetail> imageTitleList) {
```

Comment:
Remove if not needed anymore

Reply:
Yes this is not used in master anymore. Hence I have commented it down. Since there are no more usages of it, I'll remove it in my next commit